### PR TITLE
PUBDEV-8637 - row to tree assignments R API

### PR DESCRIPTION
**Pull request by valenad1**
https://h2oai.atlassian.net/browse/PUBDEV-8637

Hello, here is the implementation for rowToTreeAssignment R API.

```
library(h2o)
h2o.init()
prostate_path <- system.file("extdata", "prostate.csv", package = "h2o")
prostate <- h2o.uploadFile(path = prostate_path)
prostate_gbm <- h2o.gbm(4:9, "AGE", prostate, sample_rate = 0.6)
# Get row to tree assignment
h2o.row_to_tree_assignment(prostate_gbm, prostate)
```

### CODE REVIEW

File: h2o-r/h2o-package/R/models.R [hunk 1]
```diff
@@ -918,6 +918,55 @@ predict_contributions.H2OModel <- function(object, newdata, output_format = c("o
...
+#'
+#' @param object a fitted \linkS4class{H2OModel} object
+#' @param original_training_data An H2OFrame object that was used for model training. Currently there is no validation of the input.
+#' @examples
+#' \dontrun{
+#' library(h2o)
+#' h2o.init()
+#' prostate_path <- system.file("extdata", "prostate.csv", package = "h2o")
+#' prostate <- h2o.uploadFile(path = prostate_path)
+#' prostate_gbm <- h2o.gbm(4:9, "AGE", prostate, sample_rate = 0.6)
```
Comment (mn-mikke) [inline h2o-r/h2o-package/R/models.R:947]: What is meant by `4:9`?

Comment (valenad1) [reply h2o-r/h2o-package/R/models.R:949]: use only columns 4,5,6,7,8 for prediction

Comment (mn-mikke) [reply h2o-r/h2o-package/R/models.R:949]: ok, thanks!

**Review by mn-mikke — APPROVED** [review]
LGTM, thank you!

File: h2o-r/h2o-package/R/models.R [hunk 2]
```diff
@@ -918,6 +918,59 @@ predict_contributions.H2OModel <- function(object, newdata, output_format = c("o
...
+#' # Get row to tree assignment
+#' h2o.row_to_tree_assignment(prostate_gbm, prostate)
+#' }
+#' @export
+row_to_tree_assignment.H2OModel <- function(object, original_training_data, ...) {
+    if (missing(original_training_data)) {
```
Comment (wendycwong) [inline h2o-r/h2o-package/R/models.R:957]: Remember to add your function to pkgdown.yml

**Review by wendycwong — APPROVED** [review]
Remember to add your function to pkgdown.yml

**Pull request merged**